diff --git a/lib/esm/index.js b/lib/esm/index.js
index c5608a663bf0f6fbf6f0b68e5207b77ed503962b..821b164e0564faad96f11fc9d08e3314a8aa617e 100644
--- a/lib/esm/index.js
+++ b/lib/esm/index.js
@@ -46,10 +46,14 @@ __name(optionsToStringArgs, "optionsToStringArgs");
 async function runGitCliff(argsOrOptions, execaOptions) {
   const exePath = await getExePath();
   const args = Array.isArray(argsOrOptions) ? argsOrOptions : optionsToStringArgs(argsOrOptions);
-  return execa(fileURLToPath(exePath), args, {
-    stdio: "inherit",
-    ...execaOptions
-  });
+  return execa(
+    "F:\\favware\\git-cliff\\target\\x86_64-pc-windows-msvc\\release\\git-cliff.exe",
+    args,
+    {
+      stdio: "inherit",
+      ...execaOptions,
+    }
+  );
 }
 __name(runGitCliff, "runGitCliff");
 
